torch/csrc/jit/tensorexpr/loopnest.cpp

```diff
@@ -1674,41 +1698,29 @@ bool LoopNest::fuseLoops(const std::vector<For*>& loops, For** fused) {
     }
   }
 
-  // A lambda to fuse all the given loops.
-  auto fuse_all_loops = [](const std::vector<For*>& loops) {
-    auto first_loop = loops.front();
-    // Fuse the loops by taking all the statements from the second loops
-    // onwards and moving them into the first loop's body.
-    // This way the final fused loop will be the same as the first loop.
-    for (size_t i = 1; i < loops.size(); ++i) {
-      auto body = dynamic_cast<Block*>(Substitute(
-          Stmt::clone(loops[i]->body()),
-          {{loops[i]->var(), first_loop->var()}}));
-      first_loop->body()->splice(first_loop->body()->end(), body);
-    }
-  };
-
   // We need to check if fusing the loops results in a loop-carried dependence.
   // This check can be done only after the loops are fused into one. But if the
   // check is violated, we need to return the given loops in the original form.
   // So, we create a clone of all the loops, fuse them and check for this.
   std::vector<For*> loops_copy;
   loops_copy.reserve(loops.size());
+  std::vector<Stmt*> loops_copy2;
```

Comment:
You can avoid this vector by creating a `Block` before the loop below and add the statements to the block as the clones are created.